src/test/groovy/graphql/GraphQLTest.groovy

```diff
@@ -579,4 +581,40 @@ class GraphQLTest extends Specification {
         result == [hello: 'world']
     }
 
+    def "abort execution if query depth is too high"() {
+        given:
+        def foo = newObject()
+                .name("Foo")
+                .field(newFieldDefinition()
+                .name("field")
+                .type(new GraphQLTypeReference('Foo'))
+                .build())
+                .field(newFieldDefinition()
+                .name("scalar")
+                .type(GraphQLString)
+                .build())
+                .build()
+        GraphQLSchema schema = newSchema().query(
+                newObject()
+                        .name("RootQueryType")
+                        .field(newFieldDefinition()
+                        .name("field")
+                        .type(foo)
+                        .build()).build())
+                .build()
+
+        MaximumQueryDepthInstrumentation maximumQueryDepthInstrumentation = new MaximumQueryDepthInstrumentation(3);
+
+
+        def graphql = GraphQL.newGraphQL(schema).instrumentation(maximumQueryDepthInstrumentation).build()
+
+        when:
+        def result = graphql.execute('{ field {field {field {field {scalar}}}} }')
+
+        then:
+        result.errors.size() == 1
+        result.errors[0].message.contains("maximum query depth exceeded")
+
```

Comment:
Maybe use `where:` data driven tests and input a series of different depths.... just to be sure

Reply:
done